base/linalg/diagonal.jl

```diff
@@ -285,7 +285,7 @@ function pinv{T}(D::Diagonal{T})
 end
 function pinv{T}(D::Diagonal{T}, tol::Real)
     Di = similar(D.diag, typeof(inv(zero(T))))
-    if( !isempty(D.diag) ) maxabsD = maximum(abs(D.diag)) end
+    if( !isempty(D.diag) ) maxabsD = maximum(abs.(D.diag)) end
```

Comment:
(We should really have a `maximum(::Function, iterable)` method analogous to `sum`.  But that's a task for another PR.)